File: src/main/java/org/flossware/civilization/scenarios/StandardTechTree.java
package org.flossware.civilization.scenarios;

import org.flossware.civilization.model.Technology;

import java.util.Arrays;
import java.util.List;

/**
 * Shared technology tree used by all built-in scenarios.
 * Contains 21 technologies spanning neolithic through modern eras.
 */
public final class StandardTechTree {
    private StandardTechTree() {}

    public static List<Technology> create() {
        return Arrays.asList(
            new Technology("agriculture", "neolithic", List.of(), 50, 0.1),
            new Technology("mining", "neolithic", List.of(), 50, 0.1),
            new Technology("iron_working", "classical", List.of("mining"), 100, 0.08),
            new Technology("metallurgy_advanced", "classical", List.of("iron_working"), 150, 0.06),
            new Technology("magnetism", "classical", List.of(), 60, 0.1),
            new Technology("copper_smelting", "classical", List.of("mining"), 80, 0.09),
            new Technology("coal_mining", "medieval", List.of("mining"), 120, 0.07),
            new Technology("chemistry_basic", "medieval", List.of(), 80, 0.09),
            new Technology("steam_engine", "industrial", List.of("metallurgy_advanced", "coal_mining"), 300, 0.05),
            new Technology("combustion", "industrial", List.of("steam_engine", "chemistry_basic"), 250, 0.05),
            new Technology("electricity", "industrial", List.of("magnetism", "copper_smelting"), 350, 0.04),
            new Technology("materials_science", "modern", List.of("metallurgy_advanced"), 400, 0.03),
            new Technology("semiconductor", "modern", List.of("electricity", "materials_science"), 500, 0.02),
            new Technology("writing", "neolithic", List.of(), 40, 0.12),
            new Technology("mathematics", "classical", List.of("writing"), 90, 0.08),
            new Technology("optics", "medieval", List.of("mathematics"), 110, 0.07),
            new Technology("telescope", "industrial", List.of("optics"), 200, 0.06),
            new Technology("radio", "industrial", List.of("electricity"), 280, 0.05),
            new Technology("computing", "modern", List.of("electricity", "mathematics"), 450, 0.03),
            new Technology("internet", "modern", List.of("computing", "radio"), 550, 0.02)
        );
    }
}

File: mobile/shared/src/commonMain/kotlin/org/flossware/civilization/scenarios/StandardTechTree.kt
package org.flossware.civilization.scenarios

import org.flossware.civilization.model.Technology

object StandardTechTree {
    fun create(): List<Technology> = listOf(
        Technology("agriculture", "neolithic", emptyList(), 50.0, 0.1),
        Technology("mining", "neolithic", emptyList(), 50.0, 0.1),
        Technology("iron_working", "classical", listOf("mining"), 100.0, 0.08),
        Technology("metallurgy_advanced", "classical", listOf("iron_working"), 150.0, 0.06),
        Technology("magnetism", "classical", emptyList(), 60.0, 0.1),
        Technology("copper_smelting", "classical", listOf("mining"), 80.0, 0.09),
        Technology("coal_mining", "medieval", listOf("mining"), 120.0, 0.07),
        Technology("chemistry_basic", "medieval", emptyList(), 80.0, 0.09),
        Technology("steam_engine", "industrial", listOf("metallurgy_advanced", "coal_mining"), 300.0, 0.05),
        Technology("combustion", "industrial", listOf("steam_engine", "chemistry_basic"), 250.0, 0.05),
        Technology("electricity", "industrial", listOf("magnetism", "copper_smelting"), 350.0, 0.04),
        Technology("materials_science", "modern", listOf("metallurgy_advanced"), 400.0, 0.03),
        Technology("semiconductor", "modern", listOf("electricity", "materials_science"), 500.0, 0.02),
        Technology("writing", "neolithic", emptyList(), 40.0, 0.12),
        Technology("mathematics", "classical", listOf("writing"), 90.0, 0.08),
        Technology("optics", "medieval", listOf("mathematics"), 110.0, 0.07),
        Technology("telescope", "industrial", listOf("optics"), 200.0, 0.06),
        Technology("radio", "industrial", listOf("electricity"), 280.0, 0.05),
        Technology("computing", "modern", listOf("electricity", "mathematics"), 450.0, 0.03),
        Technology("internet", "modern", listOf("computing", "radio"), 550.0, 0.02)
    )
}
